SimpleChat.Dal/Interfaces/IRepository.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+
+namespace SimpleChat.Dal.Interfaces
+{
+    public interface IRepository<T> where T : class
```

Comment:
It's not only class it is also IDbEntity. So, you should make restriction like 
where T : class, IDbEntity